### DISCUSSION TITLE
Try adding support for WR30U, but it can not boot normally

### DISCUSSION THREAD

**Discussion by hank9999** [immortalwrt/immortalwrt#950] [Q&A]
Try adding support for WR30U, but it can not boot normally
it is stuck here and cannot boot antmore.
What i did [here](https://github.com/hank9999/immortalwrt/commit/50068aec887d00c2abece774a028805bb47b5f8f)
full boot log [wr30u-boot.log](https://gist.github.com/hank9999/9ccf7709ee42ee344b3a55b9e78d08b5)
![wr30u-boot](https://github.com/immortalwrt/immortalwrt/assets/20784591/98388d07-301d-4123-907a-5b582793ecb4)

**Comment by tisyang** [comment]
Maybe check https://github.com/hanwckf/bl-mt798x/pull/2 and https://github.com/hanwckf/immortalwrt-mt798x/pull/75 .
Are you using lean's uboot for wr30u? I have tested that lean's uboot cannot work with 360t7's op firmware, neighter 108m or vendor layout.

**Reply by hank9999** [reply]
I am using my own build u-boot actually. stock layout and ubootmod layout is available

**Comment by hank9999** [comment]
A PR has been made to the openwrt repo, this discussion will be closed
